@codex please fix

```
> dspace@3.0.0 format:check
> prettier --config .prettierrc --check "**/*.{js,ts,svelte,json,md}"

Checking formatting...
 ✓ tests/eslintPluginPresence.test.ts (1 test) 6ms
 ✓ frontend/tests/gameStateFlush.test.ts (1 test) 4ms
 ✓ tests/checkoutFetchDepth.test.ts (1 test) 4ms
 ✓ tests/npmTestRunsAllSuites.test.ts (1 test) 4ms
 ✓ tests/astroConfigLogLevel.test.ts (1 test) 10ms
 ✓ tests/prismSvelte.test.ts (1 test) 18ms
 ✓ scripts/tests/dbBenchmarkScript.test.ts (1 test) 4ms
 ✓ tests/progressionBalance.test.ts (1 test) 38ms
 ✓ tests/domException.test.ts (1 test) 3ms
 ✓ tests/indexeddb.server.test.ts (1 test) 259ms
 ✓ frontend/tests/sanity.test.ts (1 test) 3ms
[warn] src/components/__tests__/DocsIndexSearch.spec.ts
[warn] src/components/svelte/DocsIndex.svelte
[warn] Code style issues found in 2 files. Run Prettier with --write to fix.
 ❯ scripts/tests/prettierFormatting.test.ts (1 test | 1 failed) 17739ms
   × Prettier formatting > frontend sources are formatted 17737ms
     → Command failed: npm run format:check

⎯⎯⎯⎯⎯⎯⎯ Failed Tests 2 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  tests/qaDocsLinkFreshness.test.ts > QA docs test link freshness > v3.md links target the right test lines
AssertionError: Expected ../../frontend/src/components/__tests__/DocsIndexSearch.spec.ts#L69-L86 to include "combines text queries with has:image operator filters" (from v3.md:499): expected false to be true // Object.is equality

- Expected
+ Received

- true
+ false

 ❯ tests/qaDocsLinkFreshness.test.ts:62:11
     60|             doc
     61|           )}:${docLine})`
     62|         ).toBe(true);
       |           ^
     63|       }
     64|     });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/2]⎯

 FAIL  scripts/tests/prettierFormatting.test.ts > Prettier formatting > frontend sources are formatted
Error: Command failed: npm run format:check
 ❯ scripts/tests/prettierFormatting.test.ts:8:5
      6|   test('frontend sources are formatted', () => {
      7|     const frontendDir = path.join(__dirname, '../../frontend');
      8|     execSync('npm run format:check', {
       |     ^
      9|       cwd: frontendDir,
     10|       stdio: 'inherit',

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/2]⎯


 Test Files  2 failed | 285 passed (287)
      Tests  2 failed | 996 passed (998)
   Start at  07:54:54
   Duration  156.11s (transform 9.68s, setup 10.19s, collect 42.92s, tests 100.63s, environment 178.15s, prepare 32.53s)


Error: AssertionError: Expected ../../frontend/src/components/__tests__/DocsIndexSearch.spec.ts#L69-L86 to include "combines text queries with has:image operator filters" (from v3.md:499): expected false to be true // Object.is equality

- Expected
+ Received

- true
+ false

 ❯ tests/qaDocsLinkFreshness.test.ts:62:11



Error: Error: Command failed: npm run format:check
 ❯ scripts/tests/prettierFormatting.test.ts:8:5

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: { status: 1, signal: null, output: [ null, null, null ], pid: 9230, stdout: null, stderr: null }

 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

new codex task, not a r/e/v/i/e/w task

--- a/docs/qa/v3.md
+++ b/docs/qa/v3.md
@@ -496,7 +496,7 @@ Local-mode validation status (March 20, 2026):
 - [x] Docs landing search filters the manifest-driven index ([<docs-search.spec.ts:filters docs list by query and resets on clear>](../../frontend/e2e/docs-search.spec.ts#L12-L31))
 - [x] Docs search supports `has:` feature operators such as `has:link` and `has:image`
       ([<docs-search.spec.ts:supports has: feature operators for docs content>](../../frontend/e2e/docs-search.spec.ts#L31-L46),
-      [<DocsIndexSearch.spec.ts:combines text queries with has:image operator filters>](../../frontend/src/components/__tests__/DocsIndexSearch.spec.ts#L69-L86),
+      [<DocsIndexSearch.spec.ts:combines text queries with has:image operator filters>](../../frontend/src/components/__tests__/DocsIndexSearch.spec.ts#L154-L179),
       [<docsSearchFeatures.test.ts:reflects features from representative doc content>](../../tests/docsSearchFeatures.test.ts#L6-L35))
 - [x] `/docs/changelog` highlights the latest release entry and anchor
       ([<docs-changelog.spec.ts:lists recent release entries>](../../frontend/e2e/docs-changelog.spec.ts#L136))

--- a/docs/qa/v3.md
+++ b/docs/qa/v3.md
@@ -496,7 +496,7 @@ Local-mode validation status (March 20, 2026):
 - [x] Docs landing search filters the manifest-driven index ([<docs-search.spec.ts:filters docs list by query and resets on clear>](../../frontend/e2e/docs-search.spec.ts#L12-L31))
 - [x] Docs search supports `has:` feature operators such as `has:link` and `has:image`
       ([<docs-search.spec.ts:supports has: feature operators for docs content>](../../frontend/e2e/docs-search.spec.ts#L31-L46),
-      [<DocsIndexSearch.spec.ts:combines text queries with has:image operator filters>](../../frontend/src/components/__tests__/DocsIndexSearch.spec.ts#L154-L179),
+      [<DocsIndexSearch.spec.ts:combines text queries with has:image operator filters>](../../frontend/src/components/__tests__/DocsIndexSearch.spec.ts#L223-L248),
       [<docsSearchFeatures.test.ts:reflects features from representative doc content>](../../tests/docsSearchFeatures.test.ts#L6-L35))
 - [x] `/docs/changelog` highlights the latest release entry and anchor
       ([<docs-changelog.spec.ts:lists recent release entries>](../../frontend/e2e/docs-changelog.spec.ts#L136))

--- a/frontend/src/components/__tests__/DocsIndexSearch.spec.ts
+++ b/frontend/src/components/__tests__/DocsIndexSearch.spec.ts
@@ -125,6 +125,31 @@ describe('DocsIndex search operators', () => {
         expect(fetchSpy).toHaveBeenCalledTimes(1);
     });
 
+    it('uses cached body text immediately for later keyword searches', async () => {
+        const fetchSpy = vi.fn().mockImplementation(() =>
+            createFetchResponse({
+                'quest-guidelines': 'Cached turbine guidance for docs search.',
+            })
+        );
+        vi.stubGlobal('fetch', fetchSpy);
+
+        render(DocsIndex, { props: { sections: SECTIONS_FIXTURE } });
+
+        const searchBox = screen.getByRole('searchbox', { name: /search docs/i });
+
+        await fireEvent.input(searchBox, { target: { value: 'cached' } });
+        expect(
+            await screen.findByRole('link', { name: 'Quest Development Guidelines' })
+        ).not.toBeNull();
+        expect(fetchSpy).toHaveBeenCalledTimes(1);
+
+        await fireEvent.input(searchBox, { target: { value: 'guidance' } });
+        expect(
+            await screen.findByRole('link', { name: 'Quest Development Guidelines' })
+        ).not.toBeNull();
+        expect(fetchSpy).toHaveBeenCalledTimes(1);
+    });
+
     it('retries corpus loading after an initial fetch failure', async () => {
         const fetchSpy = vi
             .fn()
@@ -149,6 +174,50 @@ describe('DocsIndex search operators', () => {
             await screen.findByRole('link', { name: 'Quest Development Guidelines' })
         ).not.toBeNull();
         expect(fetchSpy).toHaveBeenCalledTimes(2);
+    });
+
+    it('treats non-ok corpus responses as retryable failures', async () => {
+        const fetchSpy = vi
+            .fn()
+            .mockResolvedValueOnce({
+                ok: false,
+                status: 503,
+                json: () => Promise.resolve({ bodyBySlug: {} }),
+            })
+            .mockImplementationOnce(() =>
+                createFetchResponse({
+                    'quest-guidelines': 'Recovered corpus content after retry.',
+                })
+            );
+        vi.stubGlobal('fetch', fetchSpy);
+
+        render(DocsIndex, { props: { sections: SECTIONS_FIXTURE } });
+
+        const searchBox = screen.getByRole('searchbox', { name: /search docs/i });
+
+        await fireEvent.input(searchBox, { target: { value: 'recoveredx' } });
+        expect(screen.queryByRole('link', { name: 'Quest Development Guidelines' })).toBeNull();
+
+        await fireEvent.input(searchBox, { target: { value: 'has:link' } });
+        await fireEvent.input(searchBox, { target: { value: 'recovered' } });
+        expect(
+            await screen.findByRole('link', { name: 'Quest Development Guidelines' })
+        ).not.toBeNull();
+        expect(fetchSpy).toHaveBeenCalledTimes(2);
+    });
+
+    it('skips corpus fetch when fetch is unavailable in the runtime', async () => {
+        vi.stubGlobal('fetch', undefined);
+
+        render(DocsIndex, { props: { sections: SECTIONS_FIXTURE } });
+
+        const searchBox = screen.getByRole('searchbox', { name: /search docs/i });
+        await fireEvent.input(searchBox, { target: { value: 'playbooks' } });
+
+        expect(screen.getByText('No docs found for "playbooks".')).not.toBeNull();
+
+        await fireEvent.input(searchBox, { target: { value: 'playbooks again' } });
+        expect(screen.getByText('No docs found for "playbooks again".')).not.toBeNull();
     });
 
     it('combines text queries with has:image operator filters', async () => {

--- a/frontend/src/components/__tests__/DocsIndexSearch.spec.ts
+++ b/frontend/src/components/__tests__/DocsIndexSearch.spec.ts
@@ -42,7 +42,10 @@ describe('DocsIndex search operators', () => {
     beforeEach(() => {
         vi.restoreAllMocks();
         vi.unstubAllGlobals();
-        vi.stubGlobal('fetch', vi.fn().mockImplementation(() => createFetchResponse()));
+        vi.stubGlobal(
+            'fetch',
+            vi.fn().mockImplementation(() => createFetchResponse())
+        );
     });
 
     it('renders default browse view from lightweight payload without fetching corpus', () => {
@@ -110,11 +113,15 @@ describe('DocsIndex search operators', () => {
         const searchBox = screen.getByRole('searchbox', { name: /search docs/i });
 
         await fireEvent.input(searchBox, { target: { value: 'turbine' } });
-        expect(await screen.findByRole('link', { name: 'Quest Development Guidelines' })).not.toBeNull();
+        expect(
+            await screen.findByRole('link', { name: 'Quest Development Guidelines' })
+        ).not.toBeNull();
         expect(fetchSpy).toHaveBeenCalledTimes(1);
 
         await fireEvent.input(searchBox, { target: { value: 'after-action' } });
-        expect(await screen.findByRole('link', { name: 'Quest Development Guidelines' })).not.toBeNull();
+        expect(
+            await screen.findByRole('link', { name: 'Quest Development Guidelines' })
+        ).not.toBeNull();
         expect(fetchSpy).toHaveBeenCalledTimes(1);
     });
 
@@ -138,7 +145,9 @@ describe('DocsIndex search operators', () => {
 
         await fireEvent.input(searchBox, { target: { value: 'has:link' } });
         await fireEvent.input(searchBox, { target: { value: 'playbooks' } });
-        expect(await screen.findByRole('link', { name: 'Quest Development Guidelines' })).not.toBeNull();
+        expect(
+            await screen.findByRole('link', { name: 'Quest Development Guidelines' })
+        ).not.toBeNull();
         expect(fetchSpy).toHaveBeenCalledTimes(2);
     });
 
@@ -192,7 +201,8 @@ describe('DocsIndex search operators', () => {
     });
 
     it('preserves long-token snippet behavior after deferred corpus loading', async () => {
-        const longToken = 'docssearchsnippettokenwithoutbreakpoints1234567890abcdefghijklmnopqrstuvwxyz';
+        const longToken =
+            'docssearchsnippettokenwithoutbreakpoints1234567890abcdefghijklmnopqrstuvwxyz';
 
         vi.stubGlobal(
             'fetch',
@@ -206,7 +216,9 @@ describe('DocsIndex search operators', () => {
         const { container } = render(DocsIndex, { props: { sections: SECTIONS_FIXTURE } });
 
         const searchBox = screen.getByRole('searchbox', { name: /search docs/i });
-        await fireEvent.input(searchBox, { target: { value: 'docssearchsnippettokenwithoutbreakpoints' } });
+        await fireEvent.input(searchBox, {
+            target: { value: 'docssearchsnippettokenwithoutbreakpoints' },
+        });
 
         await waitFor(() => expect(container.querySelector('.doc-snippet')).not.toBeNull());
         const snippet = container.querySelector('.doc-snippet');

--- a/frontend/src/components/svelte/DocsIndex.svelte
+++ b/frontend/src/components/svelte/DocsIndex.svelte
@@ -93,9 +93,11 @@
             return true;
         }
 
-        const searchableValues = [link.title, ...(link.keywords ?? []), getBodyText(link, bodyBySlug)].map(
-            normalize
-        );
+        const searchableValues = [
+            link.title,
+            ...(link.keywords ?? []),
+            getBodyText(link, bodyBySlug),
+        ].map(normalize);
 
         return (
             matchesWords(parsedQuery.keywords, searchableValues) &&
@@ -116,16 +118,15 @@
                 .filter((link) => matchLink(link, parsedQuery, bodyTextBySlug))
                 .map((link) => ({
                     ...link,
-                    snippet:
-                        parsedQuery.keywords.length
-                            ? findDocSnippet(
-                                  {
-                                      ...link,
-                                      bodyText: getBodyText(link, bodyTextBySlug),
-                                  },
-                                  parsedQuery.keywords
-                              )
-                            : null,
+                    snippet: parsedQuery.keywords.length
+                        ? findDocSnippet(
+                              {
+                                  ...link,
+                                  bodyText: getBodyText(link, bodyTextBySlug),
+                              },
+                              parsedQuery.keywords
+                          )
+                        : null,
                 })),
         }))
         .filter((section) => section.links.length > 0);

